test/CadlRanchProjects/authentication/api-key/src/Generated/Internal/ModelSerializationExtensions.cs

```diff
@@ -164,13 +165,16 @@ public static void WriteNumberValue(this Utf8JsonWriter writer, DateTimeOffset v
             writer.WriteNumberValue(value.ToUnixTimeSeconds());
         }
 
-        public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
+        public static void WriteObjectValue<T>(this Utf8JsonWriter writer, object value, ModelReaderWriterOptions options = null)
         {
             switch (value)
             {
                 case null:
                     writer.WriteNullValue();
                     break;
+                case IJsonModel<T> jsonModel:
+                    jsonModel.Write(writer, options ?? new ModelReaderWriterOptions("W"));
+                    break;
```

Comment:
I think because the interface `IJsonModel<T>` is defined as covariant of generic type, therefore I think we do not need to have this generic parameter here, just put
```csharp
case IJsonModel<object> jsonModel:
```
should just work as well.

Reply:
that would create boxing in struct case